lib/cargoAudit.nix

```diff
@@ -0,0 +1,23 @@
+{ cargoBuild
+, cargo-audit
+,
+}: { cargoArtifacts
+   , cargoAuditExtraArgs ? ""
+   , cargoExtraArgs ? ""
+   , advisory-db
+   , ...
+   } @ origArgs:
+let
+  args = builtins.removeAttrs origArgs [ "cargoAuditExtraArgs" ];
+in
+cargoBuild (args
+  // {
+  inherit cargoArtifacts;
```

Comment:
`cargo-audit` should just need the `Cargo.toml` and advisory database, right? I think it should be possible to run the check without building all the dependencies first:

- we should omit `cargoArtifacts` from the function arguments
- we should overwrite `cargoArtifacts = null` so `cargoBuild` does not try to build the dependencies internally

Reply:
`cargoArtifacts` represents _building the crate and its dependencies_ (think of it as the `./target/` directory)

`cargoVendorDir` represents making all crate dependency sources available in the builder standbox.

I think as long as we keep `cargoVendorDir` (which we are, or rather it will default to vendoring things from `Cargo.lock`) I think the scan should work, hence why I'm curious to skip the `buildDepsOnly` call as cargo-audit should be able to run without compiling anything.